app/src/main/java/io/stanwood/mhwdb/feature/armors/vm/ArmorsViewModel.kt

```diff
@@ -1,42 +1,48 @@
 package io.stanwood.mhwdb.feature.armors.vm
 
-import io.reactivex.Observable
 import io.reactivex.android.schedulers.AndroidSchedulers
 import io.reactivex.subjects.BehaviorSubject
 import io.reactivex.subjects.PublishSubject
-import io.stanwood.framework.arch.core.Resource
 import io.stanwood.framework.arch.core.ViewModel
+import io.stanwood.framework.arch.core.rx.ResourceStatusTransformer
 import io.stanwood.framework.arch.nav.NavigationTarget
 import io.stanwood.mhwdb.NavGraphMainDirections
 import io.stanwood.mhwdb.feature.armors.dataprovider.ArmorDataProvider
 import javax.inject.Inject
 
 class ArmorsViewModel @Inject constructor(private val dataProvider: ArmorDataProvider) : ViewModel {
     private val navigation = PublishSubject.create<NavigationTarget>()
-    val navigator = navigation.firstElement()
+    val navigator = navigation.firstElement()!!
     private val expandedSetsSubject = BehaviorSubject.create<MutableSet<Long>>()
     val expandedSets = mutableSetOf<Long>()
 
-    val items: Observable<List<ArmorItem>> =
+    val items =
         dataProvider.data
+            .filter { it.data != null }
```

Comment:
Shouldn't the filter/map stuff be done in the data provider?

Reply:
And provide status and data there ?
I like it this way more

Reply:
Hm, it would certainly be an alternative to have data providers serve status and data streams separately... Have to think about that a bit more. 💭